Hearthstone Deck Tracker/Hearthstone/GameV2.cs

```diff
@@ -289,19 +291,8 @@ public int GetTurnNumber()
 			return (GameEntity?.GetTag(GameTag.TURN) + 1) / 2 ?? 0;
 		}
 
-		public void SnapshotBattlegroundsBoardState()
-		{
-			var opponentHero = Entities.Values
-				.Where(x => x.IsHero && x.IsInZone(Zone.PLAY) && x.IsControlledBy(Opponent.Id))
-				.FirstOrDefault();
-			if(opponentHero == null)
-				return;
-			var entities = Entities.Values
-				.Where(x => x.IsMinion && x.IsInZone(Zone.PLAY) && x.IsControlledBy(Opponent.Id))
-				.Select(x => x.Clone())
-				.ToArray();
-			Log.Info($"Snapshotting board state for {opponentHero.Card.Name} with {entities.Length} entities");
-			LastKnownBattlegroundsBoardState[opponentHero.CardId] = new BoardSnapshot(entities, GetTurnNumber());
-		}
-	}
+		public void SnapshotBattlegroundsBoardState() => _battlegroundsBoardState.UpdateSnapshot();
+
+		public BoardSnapshot GetBattlegroundsBoardStateFor(string cardId) => _battlegroundsBoardState.GetSnapshot(cardId);
```

Comment:
I think we might as well get rid of these at this point, make `BattlegroundsBoardState` internal and access that directly. That will also make the changes to MockGame unnecessary..